src/niscope/templates/session.py/fancy_fetch_measurement_stats.py.mako

```diff
@@ -0,0 +1,29 @@
+<%page args="f, config, method_template"/>\
+<%
+    '''Forwards to _fetch_measurement_stats with cleaner return values.'''
+    import build.helper as helper
+    suffix = method_template['method_python_name_suffix']
+%>\
+    def ${f['python_name']}(${helper.get_params_snippet(f, helper.ParameterUsageOptions.SESSION_METHOD_DECLARATION)}):
+        r'''${f['python_name']}
+
+        ${helper.get_function_docstring(f, False, config, indent=8)}
+        '''
+
+        results, means, stdevs, min_vals, max_vals, nums_in_stats = self._${f['python_name']}(scalar_meas_function, timeout)
+
+        output = []
+        for result, mean, stdev, min_val, max_val, num_in_stats in zip(results, means, stdevs, min_vals, max_vals, nums_in_stats):
+            measurement_stat = measurement_stats.MeasurementStats(result, mean, stdev, min_val, max_val, num_in_stats)
+            output.append(measurement_stat)
+
+        i = 0
+        num_records = int(len(results) / len(self._repeated_capability_list))
+        for chan in self._repeated_capability_list:
+            for rec in range(0, num_records):
+                output[i].channel = chan
+                output[i].record = rec
+                i += 1
```

Comment:
Did you give some thought to facroting out duplicate logic into helper functions?
I don't think we have great mechanisms in place for this, but we can think it over if you see enough value in it.

Reply:
I thought about it briefly but wasn't sure it was worth figuring out how to add another layer of abstraction in order to save on half a dozen lines of reused code. I'd be glad to look into it, though.

Reply:
Ok. Let's look into this as a separate PR though. Go ahead and open an issue in the meantime.

Reply:
See [nidigital](https://github.com/ni/nimi-python/blob/master/src/nidigital/templates/session.py/fancy_load_specifications_levels_and_timing.py.mako) for an example of how to add a helper function. You would add it in one of the templates but could use it from multiple.

Reply:
Created #1496.